include/swift/Reflection/Records.h

```diff
@@ -110,6 +110,8 @@ enum class FieldDescriptorKind : uint16_t {
   Protocol,
```

Comment:
Some comments on the members of this enum would be nice, explaining exactly how the rest of the field descriptor is interpreted in each case.